### DISCUSSION TITLE
[31 Marzo 2021] - Modifica CPU

### DISCUSSION THREAD

**Discussion by rimaout** [CS-Swap/Architettura-degli-Elaboratori#32] [Modifica Architettura]
### Traccia Esercizio

![immagine](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/7b68adb7-3ba0-4e84-b908-06a1b407134b)
![immagine](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/48867964/5d830e71-6e31-4c6f-8e58-14990abac35f)
![immagine](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/33a07dd7-64d5-44b6-971c-e741fad0f409)

<details><summary>Soluzione Prof</summary>
<p>

![immagine](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/48867964/584f7c5f-29dc-4f9c-b961-1ffcc6ffa5d4)
![immagine](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/48867964/e6ce5fcc-f918-486d-8141-1b6ad66a6a62)
![immagine](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/48867964/f4338382-4c2a-4e37-83ce-6ea89495c7a3)

```
# Soluzione 1
add $s0, $zero, $zero       # $s0 assume valore 0
add $s1, $zero, $zero       # $s1 assume valore 0
nor $s1, $s1, $s1           # $s1 assume valore 0xFFFFFFFF
beq $s1, $s0, Exit          # In caso di guasto, il risultato di $s1-$s0 (cioè 0xFF…F) verrà scritto in $s0
Exit:

# Soluzione 2
add $s0, $zero, $zero       # $s0 assume valore 0
sw  $s0, 0xFFFFFFFF($zero)  # In caso di guasto, la somma di 0 + 0xFFFFFFFF verrà scritta in $s0

```

</p>
</details>

**Comment by alem1105** [comment]
![image](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/48867964/16559f00-b489-42ec-88e4-df66929e3317)
![image](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/48867964/8cec06ea-321d-445a-b544-f26d88a8c162)
